src/main/java/bio/terra/service/upgrade/Migrate.java

```diff
@@ -199,16 +204,20 @@ private void releaseDeploymentLock() {
         try (Connection connection = dataSource.getConnection();
              PreparedStatement statement = connection.prepareStatement(updateSql)) {
             startTransaction(connection);
-            DeploymentRow row = getDeploymentRow(connection);
-            if ((row.getLockingPodName() == null) && (migrateConfiguration.getDropAllOnStart())) {
-                // Application of the migration released the lock for us. Nothing for us to do.
-                return;
-            }
+            try {
```

Comment:
The triple-nesting of this `try ... catch` section makes it a bit difficult to reason about the code. Can we avoid one level of nesting here? Return the result of `commitReadOnlyTransaction` and then have something like:

```
} catch (SQLException ex) {
  [ ... ]
} finally {
  if(startedTransaction) {
    commitReadOnlyTransaction(connection);
  }
}
```

Reply:
Yeah, the problem is that `connection` is out of scope in the finally clause.
I split the inner part of the loop into his separate method and added the comment. See if that helps.